boards/arm/stm32f746g_disco/early_init.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2020 Mark Olsson <mark@markolsson.se>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "stm32f7xx.h"
+#include "kernel.h"
+
+// Size of early_init_vector_table at the bottom of this file
+#define VECTOR_TABLE_OFFSET 0x80;
+extern void __start(void);
+
+void configure_gpio_fmc(char port, uint8_t pin)
+{
+	/*
+	 * Alternate function: 12
+	 * Alternate function mode
+	 * Speed: 50 MHz
+	 * Output type: push-pull
+	 * No pull-up, pull-down
+	 */
+	volatile GPIO_TypeDef *GPIOx_base =
+		(void *)(GPIOA_BASE + (port - 'A') * 0x400);
+
+	if (pin >= 8) {
+		GPIOx_base->AFR[1] |= 0xC << ((pin - 8) * 4);
+	} else {
+		GPIOx_base->AFR[0] |= 0xC << (pin * 4);
+	}
+	GPIOx_base->MODER |= 0x2 << (pin * 2);
+	GPIOx_base->OSPEEDR |= 0x2 << (pin * 2);
+	GPIOx_base->OTYPER |= 0x0 << (pin * 2);
+	GPIOx_base->PUPDR |= 0x1 << (pin * 2);
```

Comment:
Using LL API would help you to make code STM32 generic.
You can find example in gpio_stm32.c

Reply:
When I include code from Zephyr or the HAL/LL api, it normally stops this from working as I don't think it can access variables stored in the .data section. I've tried to copy them, but as the linker sets things up for when the external ram is used, then trying to copy them to the on-chip ram for this early_init code, it's all wrong...often it's too big to fit also.

Any suggestions on how we can make that work?

Reply:
@ABOSTM, @FRASTM, maybe you know better how we can use/configure HAL to be used in this specific case ?

Reply:
One solution could be a small boot loader that configures the FMC...but I don't know if that will break other things?